File: implementation/scipamato/scipamato-core-entity/src/test/java/ch/difty/scipamato/core/entity/search/SearchOrderFilterTest.java
package ch.difty.scipamato.core.entity.search;

import static org.assertj.core.api.Assertions.assertThat;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.Test;

public class SearchOrderFilterTest {

    @Test
    public void canInstantiate() {
        SearchOrderFilter f = new SearchOrderFilter();
        f.setNameMask("foo");
        f.setOwner(3);
        f.setOwnerIncludingGlobal(4);
        f.setGlobal(true);

        assertThat(f.getNameMask()).isEqualTo("foo");
        assertThat(f.getOwner()).isEqualTo(3);
        assertThat(f.getOwnerIncludingGlobal()).isEqualTo(4);
        assertThat(f.getGlobal()).isEqualTo(true);

        assertThat(f.toString()).isEqualTo(
            "SearchOrderFilter(nameMask=foo, owner=3, global=true, ownerIncludingGlobal=4)");
    }

    @Test
    public void equals() {
        EqualsVerifier
            .forClass(SearchOrderFilter.class)
            .withRedefinedSuperclass()
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }
}

File: core/core-entity/src/test/kotlin/ch/difty/scipamato/core/entity/search/SearchOrderFilterTest.kt
package ch.difty.scipamato.core.entity.search

import nl.jqno.equalsverifier.EqualsVerifier
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

internal class SearchOrderFilterTest {

    @Test
    fun canInstantiate() {
        val f = SearchOrderFilter()
        f.nameMask = "foo"
        f.owner = 3
        f.ownerIncludingGlobal = 4
        f.global = true

        f.nameMask shouldBeEqualTo "foo"
        f.owner shouldBeEqualTo 3
        f.ownerIncludingGlobal shouldBeEqualTo 4
        f.global shouldBeEqualTo true

        f.toString() shouldBeEqualTo
            "SearchOrderFilter(nameMask=foo, owner=3, global=true, ownerIncludingGlobal=4)"
    }

    @Test
    fun equals() {
        EqualsVerifier.simple()
            .forClass(SearchOrderFilter::class.java)
            .withRedefinedSuperclass()
            .verify()
    }
}
